src/makim/core.py

```diff
@@ -174,7 +174,30 @@ def __init__(self) -> None:
         self.tmp_suffix: str = '.makim'
         self.scheduler = None
         # os.chdir(os.getcwd())
-
+    @property
+    def scheduler(self) -> Optional[MakimScheduler]:
+        """Lazy-initialize the scheduler."""
+        if self._scheduler is None:
+            self._scheduler = MakimScheduler(makim_instance=self)
+        return self._scheduler
+
+    @scheduler.setter
+    def scheduler(self, value: Optional[MakimScheduler]):
+        """Setter for scheduler."""
+        self._scheduler = value
+
+    def __getstate__(self):
+        """Exclude the scheduler during serialization."""
+        state = self.__dict__.copy()
+        if '_scheduler' in state:  # ✅ Target the private attribute
```

Comment:
maybe it is a personal preference .. but I would avoid using emojis in the comments